packages/compile-solidity/src/compilerSupplier/loadingStrategies/VersionRange.ts

```diff
@@ -1,20 +1,82 @@
-const debug = require("debug")("compile:compilerSupplier");
-const requireFromString = require("require-from-string");
-const fs = require("fs");
-const originalRequire = require("original-require");
-const axios = require("axios").default;
-const semver = require("semver");
-const solcWrap = require("solc/wrapper");
-const LoadingStrategy = require("./LoadingStrategy");
-
-class VersionRange extends LoadingStrategy {
+import debugModule from "debug";
+const debug = debugModule("compile:compilerSupplier");
+
+import requireFromString from "require-from-string";
+import originalRequire from "original-require";
+import axios from "axios";
+import semver from "semver";
+import solcWrap from "solc/wrapper";
+import { Cache } from "../Cache";
+import { observeListeners } from "../observeListeners";
+import { NoVersionError, NoRequestError } from "../errors";
+
+export class VersionRange {
+  private config: {
+    events: any; // represents a @truffle/events instance, which lacks types
+    compilerRoots: string[];
+  };
+
+  private cache: Cache;
+
+  constructor(options) {
+    const defaultConfig = {
+      compilerRoots: [
+        // NOTE this relay address exists so that we have a backup option in
+        // case more official distribution mechanisms fail.
+        //
+        // currently this URL just redirects (302 Found); we may alter this to
+        // host for real in the future.
+        "https://relay.trufflesuite.com/solc/bin/",
+        "https://solc-bin.ethereum.org/bin/",
+        "https://ethereum.github.io/solc-bin/bin/"
```

Comment:
@cameel brought [this](https://github.com/trufflesuite/truffle/pull/4222#discussion_r689001206) to our attention and we need to make sure we've got this information properly processed.

Reply:
Seems like the entire comment suffices as an issue, which I've opened [here](https://github.com/trufflesuite/truffle/issues/4425)